tiledb/sm/misc/parallel_functions.h

```diff
@@ -232,7 +273,13 @@ Status parallel_for(
   // Wait for all instances of `execute_subrange` to complete.
   tp->wait_all(tasks);
```

Comment:
Per our slack exchange -- we could also just get the return `Status` here -- though it would change the behavior.